tests/unit/config.py

```diff
@@ -25,3 +27,161 @@ def test_load_extension():
     config.load_extensions()
     for module in 'sndhdr', 'colorsys', 'email.mime.audio':
         assert module in sys.modules
+
+
+def test_validate_config():
+    """Test validing a HIL config file."""
+    config_testsuite()
+    config_merge({
+        'general': {
+            'log_level': 'debug',
+        },
+        'auth': {
+            'require_authentication': 'True',
+        },
+        'headnode': {
+            'trunk_nic': 'eth0',
+            'libvirt_endpoint': 'qemu:///system',
+        },
+        'client': {
+            'endpoint': 'http://127.0.0.1:5000',
+        },
+        'maintenance': {
+            'maintenance_project': 'maintenance',
+            'url': 'http://test.xyz',
+            'shutdown': '',
+        },
+        'extensions': {
+            'hil.ext.switches.dell': '',
+            'hil.ext.switches.dellnos9': '',
+            'hil.ext.switches.brocade': '',
+            'hil.ext.switches.n3000': '',
+            'hil.ext.switches.nexus': '',
+            'hil.ext.auth.null': None,
+            'hil.ext.auth.keystone': '',
+            'hil.ext.network_allocators.null': None,
+            'hil.ext.network_allocators.vlan_pool': '',
+        },
+        'hil.ext.network_allocators.vlan_pool': {
+            'vlans': '12, 13-20, 100-200',
+        },
+        'hil.ext.switches.dell': {
+            'save': 'True',
+        },
+        'hil.ext.switches.dellnos9': {
+            'save': 'False',
+        },
+        'hil.ext.switches.brocade': {
+            'save': 'True',
+        },
+        'hil.ext.switches.n3000': {
+            'save': 'False',
+        },
+        'hil.ext.switches.nexus': {
+            'save': 'True',
+        },
+        'hil.ext.auth.keystone': {
+            'auth_url': 'https://website:35357/v3',
+            'auth_protocol': 'http',
+            'username': 'admin',
+            'password': 's3cr3t',
+            'project_name': 'project',
+            'admin_user': 'admin',
+            'admin_password': 's3cr3t',
+        }
+    })
+    config.load_extensions()
+    config.validate_config()
+
+
+def test_bool_validation_upper():
+    """Test valid upper-case bools."""
+    opts = ['True', 'Yes', 'On', '1', 'False', 'No', 'Off', '0']
+    assert all(config.string_is_bool(s) for s in opts)
+
+
+def test_bool_validation_lower():
+    """Test valid lower-case bools."""
+    opts = ['true', 'yes', 'on', '1', 'false', 'no', 'off', '0']
+    assert all(config.string_is_bool(s) for s in opts)
+
+
+def test_bool_validation_bad_cases():
+    """Test invalid bools."""
+    opts = ['frue', 'nes', 'yOn', '2', 'Salse', 'Go', 'bff', '3']
+    for s in opts:
+        with pytest.raises(SchemaError):
+            config.string_is_bool(opts)
```

Comment:
Think this should `config.string_is_bool(s)`. Yay for linters catching real bugs.

Reply:
Ah so that's where it was! I was having a hard time figuring that out last week,